### PR TITLE
Implement the call to the Fastly API

### PR DESCRIPTION
**Pull request by cb372**
For every delete event from Crier, the Lambda will send a hard purge request to the Fastly API.

Note: the call to the API is untested, as I can't test it without purging stuff, obvs.

@johnduffell

### CODE REVIEW

**Comment by cb372** [comment]
The Lambda is set up in the CAPI account and ready to go, but I haven't hooked it up to the Crier stream so it won't do anything.

**Comment by LATaylor-guardian** [comment]
Very happy to see this!

File: src/main/scala/com/gu/fastly/Lambda.scala [hunk 1]
```diff
@@ -1,30 +1,61 @@
...
+import com.gu.crier.model.event.v1._
 import scala.collection.JavaConverters._
 
 class Lambda {
 
   private val config = Config.load()
   private val httpClient = new OkHttpClient()
 
-  def recordHandler(event: KinesisEvent) {
+  def handle(event: KinesisEvent) {
```
Comment (LATaylor-guardian) [inline src/main/scala/com/gu/fastly/Lambda.scala]: Have you updated the handler name in the Lambda setup given this change?

File: src/main/scala/com/gu/fastly/Lambda.scala [hunk 2]
```diff
@@ -1,30 +1,61 @@
...
 
     CrierEventProcessor.process(userRecords.asScala) { event =>
-      // TODO calculate surrogate key (md5hex of path)
-      // TODO if it's a delete event, send purge request to Fastly
-
-      true
+      (event.itemType, event.eventType) match {
+        case (ItemType.Content, EventType.Delete) =>
+          sendPurgeRequest(event.payloadId)
+        case other =>
```
Comment (LATaylor-guardian) [inline src/main/scala/com/gu/fastly/Lambda.scala]: Can this just be _ seeing as your not using the named variable?

**Comment by johnduffell** [comment]
yep the purging bit looks great to me... also if it works well we can add soft purging for non-delete events and get rid of the job in frontend admin. 👍 (from frontend) and thanks!
Still seems weird that we have to run the lambda in the same account as the stream though.

**Comment by LATaylor-guardian** [comment]
@johnduffell Yeah, it's unavoidable at current and has been causing quite a bit of pain and duplication for us and others!

File: src/main/scala/com/gu/fastly/Lambda.scala (hunk 1 above)
Comment (cb372) [inline src/main/scala/com/gu/fastly/Lambda.scala]: Yes. Thanks for spotting it though.

File: src/main/scala/com/gu/fastly/Lambda.scala (hunk 2 above)
Comment (cb372) [inline src/main/scala/com/gu/fastly/Lambda.scala]: Yep, could be but I won't bother changing it

**Comment by cb372** [comment]
I've deployed this version of the Lambda and hooked it up to Crier's Kinesis stream. Let's see how it goes!

**Comment by LATaylor-guardian** [comment]
@cb372 👍

**Pull request merged**